packages/framework/memo-cache/src/getCacheEntry.ts

```diff
@@ -0,0 +1,30 @@
+export type CacheEntry<T, TGet = any> = {
+  value?: T;
+  empty?: CacheEntry<TGet>;
+  noargs?: CacheEntry<TGet>;
+  str?: { [key: string]: CacheEntry<TGet> };
+  obj?: WeakMap<object, TGet>;
+};
+
+function ensureAndReturn(entry: CacheEntry<any>, key: keyof CacheEntry<any>): CacheEntry<any> | { [key: string]: CacheEntry<any> } {
+  return (entry[key] = entry[key] || {});
+}
+
+function jumpToCacheEntry(entry: CacheEntry<any>, val: any): CacheEntry<any> {
```

Comment:
This code is fairly internal and hidden, so it shouldn't matter too much, but it could benefit from comments or at least less reliance on chaining assignment operators. It is somewhat difficult to read as is.

Reply:
Good point, put a bunch of comments in.